api/src/tests/simulation_test.rs

```diff
@@ -32,32 +33,38 @@ async fn simulate_aptos_transfer(
         let signature = use_valid_signature
             .then(|| signature.to_string())
             .unwrap_or(Ed25519Signature::dummy_signature().to_string());
-        context
-            .expect_status_code(expected_status)
-            .post(
-                "/transactions/simulate",
-                json!({
-                    "sender": txn.sender().to_string(),
-                    "sequence_number": txn.sequence_number().to_string(),
-                    "max_gas_amount": txn.max_gas_amount().to_string(),
-                    "gas_unit_price": txn.gas_unit_price().to_string(),
-                    "expiration_timestamp_secs": txn.expiration_timestamp_secs().to_string(),
-                    "payload": {
-                        "type": "entry_function_payload",
-                        "function": "0x1::aptos_account::transfer",
-                        "type_arguments": [],
-                        "arguments": [
-                            bob.address().to_standard_string(), transfer_amount.to_string(),
-                        ]
-                    },
-                    "signature": {
-                        "type": "ed25519_signature",
-                        "public_key": public_key.to_string(),
-                        "signature": signature,
-                    }
-                }),
-            )
-            .await
+        let req = warp::test::request()
+            .method("POST")
+            .path("/v1/transactions/simulate")
+            .json(&json!({
+                "sender": txn.sender().to_string(),
+                "sequence_number": txn.sequence_number().to_string(),
+                "max_gas_amount": txn.max_gas_amount().to_string(),
+                "gas_unit_price": txn.gas_unit_price().to_string(),
+                "expiration_timestamp_secs": txn.expiration_timestamp_secs().to_string(),
+                "payload": {
+                    "type": "entry_function_payload",
+                    "function": "0x1::aptos_account::transfer",
+                    "type_arguments": [],
+                    "arguments": [
+                        bob.address().to_standard_string(), transfer_amount.to_string(),
+                    ]
+                },
+                "signature": {
+                    "type": "ed25519_signature",
+                    "public_key": public_key.to_string(),
+                    "signature": signature,
+                }
+            }));
+        let resp = context.expect_status_code(expected_status).reply(req).await;
+        // Assert the gas used header is present if expected.
+        if let Some(expected_gas_used) = expected_gas_used {
+            assert_eq!(
+                resp.headers().get("X-Aptos-Gas-Used").unwrap(),
+                expected_gas_used.to_string().as_str()
+            );
```

Comment:
Yeah this looks a bit fragile to me, since the gas cost of a transaction could change if we modify or re-calibrate the gas schedule.

Would it make sense if you just assert that `resp.headers().get("X-Aptos-Gas-Used").is_some()`?

Reply:
I had that at first actually because I also thought it was fragile, but I wanted a better test. How about we assert gas is at least > 0?